code/modules/mob/living/carbon/alien/special/alien_embryo.dm

```diff
@@ -7,24 +7,27 @@
 	icon_state = "larva0_dead"
 	var/stage = 0
 	var/polling = FALSE
+	var/incubation_time_per_stage = 70 SECONDS //How long it takes for an alien embryo to advance a stage in it's development// Pretend this is 70 seconds it basically is I swear
+	var/incubation_deviation = 0 //The random deviation for how long the incubation period per stage will take, ranging from -15% to +15. NOTE! If you have a better name for this var, I'd love it
+	var/last_stage_progress = 0 //Used to keep track of when incubation progressed to the next stage
 
 /obj/item/organ/internal/body_egg/alien_embryo/on_find(mob/living/finder)
 	..()
 	if(stage < 4)
 		to_chat(finder, "It's small and weak, barely the size of a fetus.")
 	else
 		to_chat(finder, "It's grown quite large, and writhes slightly as you look at it.")
-		if(prob(10))
-			AttemptGrow(burst_on_success = FALSE)
+		AttemptGrow(burst_on_success = FALSE)
 
 /obj/item/organ/internal/body_egg/alien_embryo/prepare_eat()
 	var/obj/S = ..()
 	S.reagents.add_reagent("sacid", 10)
 	return S
 
-/obj/item/organ/internal/body_egg/alien_embryo/on_life()
+/obj/item/organ/internal/body_egg/alien_embryo/on_life() //I'm gonna do something about this, I swear
+	incubation_deviation = PERCENT_OF(rand(85, 115), incubation_time_per_stage) //The actual deviation location, and where the magic happens
```

Comment:
Do we need recalculate this every cycle? I would move it to where the stage advances. Before/after the `last_stage_progress = world.time` line for example.

Reply:
I agree with steel here

Reply:
Addressed!